src/Compilers/CSharp/Portable/CommandLine/CommandLineParser.cs

```diff
@@ -898,6 +899,27 @@ public new CSharpCommandLineArguments Parse(IEnumerable<string> args, string bas
                             printFullPaths = true;
                             continue;
 
+                        case "pathmap":
+                            // "/pathmap:K1=V1,K2=V2..."
+                            {
+                                if (value == null)
+                                    break;
+
+                                var pathMapBuilder = ArrayBuilder<KeyValuePair<string, string>>.GetInstance();
+                                foreach (var kEqualsV in value.Split(','))
+                                {
+                                    var kv = kEqualsV.Split('=');
+                                    if (kv.Length != 2) continue;
+                                    var from = TrimTrailingSeparators(kv[0]);
+                                    var to = TrimTrailingSeparators(kv[1]);
+                                    if (from.Length == 0 || to.Length == 0) continue;
+                                    pathMapBuilder.Add(new KeyValuePair<string, string>(from, to));
+                                }
+
+                                pathMap = pathMapBuilder.ToImmutableAndFree();
```

Comment:
Yes, I would prefer an error message in this case. I am not sure whether this is consistent with other switches. If for other switches the last value wins, I guess doing the same for the new switch is acceptable. Regarding the duplicate keys, that should be an error, or, again, the last should win for consistency.